src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -2632,14 +2632,14 @@ A catch() block after a catch (System.Exception e) block can catch non-CLS excep
   <data name="ERR_ForEachMissingMember" xml:space="preserve">
     <value>foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'</value>
   </data>
-  <data name="ERR_AsyncForEachMissingMember" xml:space="preserve">
+  <data name="ERR_AwaitForEachMissingMember" xml:space="preserve">
     <value>Async foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'</value>
   </data>
   <data name="ERR_ForEachMissingMemberWrongAsync" xml:space="preserve">
-    <value>foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'. Did you mean 'foreach await'?</value>
+    <value>foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'. Did you mean 'await foreach'?</value>
   </data>
-  <data name="ERR_AsyncForEachMissingMemberWrongAsync" xml:space="preserve">
-    <value>Async foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'. Did you mean 'foreach' rather than 'foreach await'?</value>
+  <data name="ERR_AwaitForEachMissingMemberWrongAsync" xml:space="preserve">
+    <value>Async foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'. Did you mean 'foreach' rather than 'await foreach'?</value>
```

Comment:
This reads really strange that an "async foreach" is "await foreach".

Reply:
Took a follow-up note in work-items: https://github.com/dotnet/roslyn/issues/24037
The message should be "`await foreach` ..." or "Asynchronous foreach ..."
Thanks